### PR TITLE
Denovodesign

### PR DESCRIPTION
**Pull request by mrwns**
Here I propose a few edits/additions to the de novo design section in "treat", and a few references I added in the ligand based virtual screening section.

### CODE REVIEW

**Comment by mrwns** [comment]
I removed the discussion about 2d vs 3d representations again, as it might be going too much into detail here.

File: sections/05_treat.md [hunk 1]
```diff
@@ -224,12 +224,12 @@ how drugs are metabolized [@doi:10.1021/ci400518g].
...
 features without requiring prior knowledge of the target. Chemical features may
 be represented as a list of molecular descriptors such as molecular weight, atom
-counts, charge representations, summaries of atom-atom relationships in the
+counts, functional groups, charge representations, summaries of atom-atom relationships in the
 molecular graph, and more sophisticated derived properties
 [@doi:10.1002/9783527628766].   Alternatively, chemicals can be characterized
 with the fingerprint bit vectors, textual strings, or novel learned
 representations described below. Neural networks have a long history in this
-domain [@tag:Baskin2015_drug_disc], and the 2012 Merck Molecular Activity
+domain [@tag:Baskin2015_drug_disc][@doi:10.1002/minf.201501008], and the 2012 Merck Molecular Activity
```
Comment (agitter) [inline sections/05_treat.md]: Agreed, this is good to add.  We have these two and another two drug discovery-related reviews in the intro.

We can merge the refs: `[@tag:Baskin2015_drug_disc @doi:10.1002/minf.201501008]`

File: sections/05_treat.md [hunk 2]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
 learning for drug discovery are *de novo* drug design and protein
-structure-based models.  Whereas the goal of virtual screening is to prediction
-the biochemical activity of hundreds of thousands to millions of chemicals, *de
-novo* drug design explores the much larger space of at least 10<sup>60</sup>
-organic molecules with drug-like properties that could be chemically synthesized
-[@doi:10.1002/wcms.1104].  Generative neural networks for drug design typically
+structure-based models. Whereas the goal of virtual screening is to 
+find active molecules by predicting the biochemical activity of hundreds of
+thousands to millions of chemicals using existing (virtual) collections of molecules, 
+analogous to robot based high throughput "wet lab" screening, _de novo_ drug design
```
Comment (agitter) [inline sections/05_treat.md]: "high-throughput"

File: sections/05_treat.md [hunk 3]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
-structure-based models.  Whereas the goal of virtual screening is to prediction
-the biochemical activity of hundreds of thousands to millions of chemicals, *de
-novo* drug design explores the much larger space of at least 10<sup>60</sup>
-organic molecules with drug-like properties that could be chemically synthesized
-[@doi:10.1002/wcms.1104].  Generative neural networks for drug design typically
+structure-based models. Whereas the goal of virtual screening is to 
+find active molecules by predicting the biochemical activity of hundreds of
+thousands to millions of chemicals using existing (virtual) collections of molecules, 
+analogous to robot based high throughput "wet lab" screening, _de novo_ drug design 
+aims to directly _generate_ active compounds.[@doi:10.1002/wcms.49,@doi:10.1021/acs.jmedchem.5b01849]
```
Comment (agitter) [inline sections/05_treat.md]: Replace comma with space between references and move before period.

File: sections/05_treat.md [hunk 4]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+
+It attempts to model the typical design-synthesize-test cycle of drug discovery.[@doi:10.1002/wcms.49]
```
Comment (agitter) [inline sections/05_treat.md]: To lead a paragraph, I suggest "Drug design...".

Also, move the reference.

File: sections/05_treat.md [hunk 5]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
-organic molecules with drug-like properties that could be chemically synthesized
-[@doi:10.1002/wcms.1104].  Generative neural networks for drug design typically
+structure-based models. Whereas the goal of virtual screening is to 
+find active molecules by predicting the biochemical activity of hundreds of
+thousands to millions of chemicals using existing (virtual) collections of molecules, 
+analogous to robot based high throughput "wet lab" screening, _de novo_ drug design 
+aims to directly _generate_ active compounds.[@doi:10.1002/wcms.49,@doi:10.1021/acs.jmedchem.5b01849]
+
+It attempts to model the typical design-synthesize-test cycle of drug discovery.[@doi:10.1002/wcms.49] 
+Thus de novo design explores in principle without explicit enumeration the much larger space
```
Comment (agitter) [inline sections/05_treat.md]: *de novo*

File: sections/05_treat.md [hunk 6]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
-[@doi:10.1002/wcms.1104].  Generative neural networks for drug design typically
+structure-based models. Whereas the goal of virtual screening is to 
+find active molecules by predicting the biochemical activity of hundreds of
+thousands to millions of chemicals using existing (virtual) collections of molecules, 
+analogous to robot based high throughput "wet lab" screening, _de novo_ drug design 
+aims to directly _generate_ active compounds.[@doi:10.1002/wcms.49,@doi:10.1021/acs.jmedchem.5b01849]
+
+It attempts to model the typical design-synthesize-test cycle of drug discovery.[@doi:10.1002/wcms.49] 
+Thus de novo design explores in principle without explicit enumeration the much larger space 
+of an estimated 10^60 synthesizable organic molecules with drug-like properties.[@doi:10.1002/wcms.1104]
```
Comment (agitter) [inline sections/05_treat.md]: `10<sup>60</sup>` works for superscripts.  Move reference.

File: sections/05_treat.md [hunk 7]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+To test or score structures, machine-learning algorithms as those discussed earlier
```
Comment (agitter) [inline sections/05_treat.md]: "as" to "like"

File: sections/05_treat.md [hunk 8]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+ are used. To "design" and "synthesize", traditional de-novo design software relied on classical optimizers such as genetic
```
Comment (agitter) [inline sections/05_treat.md]: *de novo*

File: sections/05_treat.md [hunk 9]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+ algorithms. Unfortunately, this often leads to overfitted, 'weird' molecules which are difficult to synthesize in the lab.
```
Comment (agitter) [inline sections/05_treat.md]: Double quotes

", which"

File: sections/05_treat.md [hunk 10]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+aims to directly _generate_ active compounds.[@doi:10.1002/wcms.49,@doi:10.1021/acs.jmedchem.5b01849]
+
+It attempts to model the typical design-synthesize-test cycle of drug discovery.[@doi:10.1002/wcms.49] 
+Thus de novo design explores in principle without explicit enumeration the much larger space 
+of an estimated 10^60 synthesizable organic molecules with drug-like properties.[@doi:10.1002/wcms.1104]
+To test or score structures, machine-learning algorithms as those discussed earlier
+ are used. To "design" and "synthesize", traditional de-novo design software relied on classical optimizers such as genetic
+ algorithms. Unfortunately, this often leads to overfitted, 'weird' molecules which are difficult to synthesize in the lab. 
+To generate molecules, current programs have therefore settled on rule-based virtual chemical 
+reactions to generate molecular structures.[@doi:10.1021/acs.jmedchem.5b01849]
```
Comment (agitter) [inline sections/05_treat.md]: Move reference

File: sections/05_treat.md [hunk 11]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+
+Neural network models that learn to generate realistic, synthesizable molecules have been proposed
+ as an alternative to provide the large molecule sets needed for
+ virtual screening or even create and refine focussed molecules for de novo design.
```
Comment (agitter) [inline sections/05_treat.md]: *de novo*

File: sections/05_treat.md [hunk 12]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+ are used. To "design" and "synthesize", traditional de-novo design software relied on classical optimizers such as genetic
+ algorithms. Unfortunately, this often leads to overfitted, 'weird' molecules which are difficult to synthesize in the lab. 
+To generate molecules, current programs have therefore settled on rule-based virtual chemical 
+reactions to generate molecular structures.[@doi:10.1021/acs.jmedchem.5b01849]
+
+Neural network models that learn to generate realistic, synthesizable molecules have been proposed
+ as an alternative to provide the large molecule sets needed for
+ virtual screening or even create and refine focussed molecules for de novo design. 
+In contrast to the classical, symbolic approaches, generative models learned from data do not depend on laboriously encoded expert knowledge.
+ The problem is related to the generation of syntactically and semantically correct text.[@arxiv:1308.0850]
```
Comment (agitter) [inline sections/05_treat.md]: Move reference

File: sections/05_treat.md [hunk 13]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+
+As neural models that directly output (molecular) graphs remain under-explored,
```
Comment (agitter) [inline sections/05_treat.md]: "neural network"

File: sections/05_treat.md [hunk 14]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+ generative neural networks for drug design typically
 represent chemicals with the simplified molecular-input line-entry system
 (SMILES), a standard way string-based representation with characters that
-represent atoms, bonds, and rings [@tag:Segler2017_drug_design].
+represent atoms, bonds, and rings [@tag:Segler2017_drug_design]. 
+This allows treating molecules as sequences and thus to leverage recent progress in
```
Comment (agitter) [inline sections/05_treat.md]: "and leveraging"

File: sections/05_treat.md [hunk 15]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+ generative neural networks for drug design typically
 represent chemicals with the simplified molecular-input line-entry system
 (SMILES), a standard way string-based representation with characters that
-represent atoms, bonds, and rings [@tag:Segler2017_drug_design].
+represent atoms, bonds, and rings [@tag:Segler2017_drug_design]. 
+This allows treating molecules as sequences and thus to leverage recent progress in 
+recurrent neural networks.
 Gómez-Bombarelli et al. designed a SMILES-to-SMILES autoencoder to learn a
 continuous latent feature space for chemicals [@tag:Gomezb2016_automatic]. In
 this learned continuous space it was possible to train some types of supervised
```
Comment (agitter) [inline sections/05_treat.md]: Can cut the supervised learning phrase because you describe them below

File: sections/05_treat.md [hunk 16]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+within this latent space. The strategy of constructing simple, continuous features
 before applying supervised learning techniques is reminiscent of autoencoders
-trained on high-dimensional EHR data [@tag:BeaulieuJones2016_ehr_encode].  A
+trained on high-dimensional EHR data [@tag:BeaulieuJones2016_ehr_encode]. A
 drawback of the SMILES-to-SMILES autoencoder is that not all SMILES strings
 produced by the autoencoder's decoder correspond to valid chemical structures.
-More recent approaches train RNNs on compounds from ChEMBL
+Recently, the Grammar Variational Autoencoder, which takes the SMILES grammar into account
+ and is guaranteed to produce syntactically valid SMILES,
+ has been proposed to alleviate this issue.[@arxiv:1703.01925]
```
Comment (agitter) [inline sections/05_treat.md]: I hadn't seen this paper, very cool.

Move the reference.

File: sections/05_treat.md [hunk 17]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+Another approach is to train character-based RNNs on large collections of molecules, for example ChEMBL
```
Comment (agitter) [inline sections/05_treat.md]: Add comma

File: sections/05_treat.md [hunk 18]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
-@tag:Olivecrona2017_drug_design].  These generative models successfully learn
+drug-like compounds [@tag:Segler2017_drug_design]. These generative models successfully learn
 the grammar of compound representations, with 94%
 [@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of
 generated SMILES corresponding to valid molecular structures.  The initial RNN
 is then fine-tuned to generate molecules that are likely to be active against a
 specific target by either continuing training on a small set of positive
 examples [@tag:Segler2017_drug_design] or adopting reinforcement learning
-strategies [@tag:Olivecrona2017_drug_design].
+strategies [@tag:Olivecrona2017_drug_design][@arxiv:1611.02796].
```
Comment (agitter) [inline sections/05_treat.md]: Merge references

File: sections/05_treat.md [hunk 19]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+drug-like compounds [@tag:Segler2017_drug_design]. These generative models successfully learn
 the grammar of compound representations, with 94%
 [@tag:Olivecrona2017_drug_design] or nearly 98% [@tag:Segler2017_drug_design] of
 generated SMILES corresponding to valid molecular structures.  The initial RNN
 is then fine-tuned to generate molecules that are likely to be active against a
 specific target by either continuing training on a small set of positive
 examples [@tag:Segler2017_drug_design] or adopting reinforcement learning
-strategies [@tag:Olivecrona2017_drug_design].
+strategies [@tag:Olivecrona2017_drug_design][@arxiv:1611.02796]. 
+Both fine-tuning and reinforcement learning strategies could rediscover known, hold-out active molecules.
```
Comment (agitter) [inline sections/05_treat.md]: "held out"

File: sections/05_treat.md [hunk 20]
```diff
@@ -373,35 +373,63 @@ specialized subsets.
...
+The great flexibility of neural networks, and progress in generative models offers many opportunites for 
+deep architectures in de novo design, for example, the adaptation of Generative Adversarial Networks (GANs) for molecules.
```
Comment (agitter) [inline sections/05_treat.md]: *de novo*

**Comment by agitter** [comment]
@mrwns Thanks for these revisions.  Please feel welcome to add yourself as an author to `author_contact_approve.md` if you read and approve of the rest of the manuscript.

This section has gotten quite long now and could use sub-headers.  Right now `Ligand-Based Prediction of Bioactivity` is the only header, but that only applies to the first part of the section.  I'll submit a new pull request with additional sub-sections that you can review.

Travis is hanging so if there are build errors I'll fix them in my next pull request for this section.

**Comment by mrwns** [comment]
Great! Thanks @agitter !

**Pull request merged**